fdbserver/VersionedBTree.actor.cpp

```diff
@@ -35,6 +35,12 @@
 #include <string.h>
 #include "flow/actorcompiler.h"
 
+#ifdef NO_INTELLISENSE
+#define THIS_PTR this
+#else
+#define THIS_PTR nullptr
+#endif
```

Comment:
Probably - I only saw very few occurrences where this is needed. Which is why I didn't want to put it in `flow.h`. Furthermore I am not sure whether having these in `flow.h` would break something else.

Reply:
There seem to more occurrences of `this` within an ACTOR in this file, but they're only compiled in if -DREDWOOD_DEBUG=1. I think it makes sense to change those too

Reply:
actorcompiler.h instead of flow.h then?  Overall, this seems like something that you would need to do in any actor in which you try to use `this` in a lambda.  So that should become a rule documented somewhere, and an official list of macros to work around the issues should be available from some file?

Reply:
I moved `THIS` and `THIS_ADDR` to `actorcompiler.h` and undefined them in `unactorcompiler.h`. It seems to me that it would be unsafe to define these in `flow.h` as they might be defined in 3rd party libraries as well. However, this should work well. I also got rid of the evil C-style cast.

@atn34 I also fixed all those references.